src/main/clojure/clara/rules/durability.clj

```diff
@@ -372,23 +417,41 @@
                                   it (.iterator es)]
                               (when (.hasNext it)
                                 (loop [^java.util.Map$Entry e (.next it)]
-                                  (aset arr (:idx (.getValue e)) ^Object (.getKey e))
+                                  (let [pair (map-entry->arr-idx-pair e)]
+                                    (aset arr (nth pair 0) (nth pair 1)))
                                   (when (.hasNext it)
                                     (recur (.next it)))))
                               (into [] arr)))
 
+        idx-fact-arr-pair-fn (fn [^java.util.Map$Entry e]
```

Comment:
I think it'd be easier to read if you defined these 2 "pair-fn"s above the `vec-indexed-facts` that uses them as a parameter.